### PR TITLE
feat: set and hook up libp2p-connection-manager

### PR DESCRIPTION
**Pull request by pgte**
Solves #183

### CODE REVIEW

**Comment by pgte** [comment]
@diasdavid I'm not sure about [using `options.connectionManager`](https://github.com/libp2p/js-libp2p/pull/184/files#diff-1fdf421c05c1140f6d71444ea2b27638R41). Since you're more familiar with the libp2p API, what do you think?

**Comment by VictorBjelkholm** [comment]
`error Couldn't find package "libp2p-connection-manager" on the "npm" registry.` errors all over CI

**Comment by pgte** [comment]
@VictorBjelkholm that's expected, it's not released yet and I can't do that. @diasdavid ?

**Comment by pgte** [comment]
@diasdavid ping

**Comment by diasdavid** [comment]
@pgte just released it for you and added you as an owner. I'm intrigued on why you weren't able to do it? It was the first time the module got published -- https://www.npmjs.com/package/libp2p-connection-manager.

**Comment by pgte** [comment]
@diasdavid I never did ipfs or libp2p releases, it would have been the first time, so I would have needed the go-ahead. I've seen you added me as an npm owner. Thank you!

**Comment by diasdavid** [comment]
CI is still red. Any ETA to get this ready for prime time?

**Comment by pgte** [comment]
Updated the libp2p-connection-manager dev version, but the CI is still breaking, now with the errors:

TravisCI:

```
Error: Unable to find native module
    at requireNativeModule (/home/travis/build/libp2p/js-libp2p/node_modules/node-cmake/index.js:40:20)
    at Object.<anonymous> (/home/travis/build/libp2p/js-libp2p/node_modules/wrtc/lib/binding.js:3:39)
    at Module._compile (internal/modules/cjs/loader.js:678:30)
```

Jenkins:

```
  1) stream muxing spdy + mplex:
     Uncaught Error: Unhandled "error" event. (undefined)
      at node_modules/spdy-transport/lib/spdy-transport/connection.js:757:12
      at Array.forEach (<anonymous>)
      at Connection.destroyStreams (node_modules/spdy-transport/lib/spdy-transport/connection.js:753:33)
```

and others...

I don't think they're related to these changes. @VictorBjelkholm @diasdavid can you help?

**Comment by VictorBjelkholm** [comment]
@pgte I rerun the tests and now they are passing. I would add a comment above the test-case, noting that it's flaky.

**Comment by diasdavid** [comment]
@pgte could you describe what happens with the connection manager enabled?

**Comment by pgte** [comment]
@diasdavid since the defaults impose no threshold, nothing should happen.

**Comment by alanshaw** [comment]
This is awesome! How would someone pass connection manager options to js-ipfs to enable the limits? Do we need a PR to js-ipfs to allow it?

We should also document the options in js-ipfs (or at least link to the [relevant section on the connection manager repo](https://github.com/libp2p/js-libp2p-connection-manager#create-a-connectionmanager)).

Also, is there something left to do on this PR?

**Comment by pgte** [comment]
I think this is working, but I'd like to get some consensus around how the connection manager options are passed in. Right now, it's from `options.connectionManager`. Sounds good?

**Comment by alanshaw** [comment]
Looks sensible to me

**Comment by pgte** [comment]
Awesome. Then this should be all! :)

**Comment by diasdavid** [comment]
CI is red. Can we get it green?

**Comment by alanshaw** [comment]
Ok so, I've fixed the merge conflict, which brings in updated dependencies. We're now getting the _exact same_ test failures in CI that I'm seeing in master (I've [logged them here](https://github.com/libp2p/js-libp2p/pull/205) also) so I think they might be unrelated to the changes in this PR.

File: package.json [hunk 1]
```diff
@@ -62,15 +63,15 @@
...
     "libp2p-websocket-star": "~0.8.0",
     "libp2p-websocket-star-rendezvous": "~0.2.3",
     "lodash.times": "^4.3.2",
     "pull-goodbye": "0.0.2",
     "pull-serializer": "~0.3.2",
     "pull-stream": "^3.6.8",
     "sinon": "^5.0.7",
-    "libp2p-webrtc-star": "~0.15.0",
-    "wrtc": "0.1.1"
+    "wrtc": "~0.1.4"
```
[the remark this answers is not in the document]
Comment (alanshaw) [reply package.json]: Is there a bug report open on the wrtc repo?

**Comment by alanshaw** [comment]
aha, tests have now passed on Node.js 8 but fail on Node.js 10 because there's no prebuilt binary for `wrtc`. I think we good to merge? 💣 💥

**Pull request merged**